VariantShredding.md

```diff
@@ -25,290 +25,316 @@
 
-Similarly the elements of an `array` must be a group containing one or more of `object`, `array`, `typed_value` or `variant_value`.
+An object is _partially shredded_ when the `value` is an object and the `typed_value` is a shredded object.
 
-Each leaf in the schema can store an arbitrary Variant value.
-It contains an `variant_value` binary field and a `typed_value` field.
-If non-null, `variant_value` represents the value stored as a Variant binary.
-The `typed_value` field may be any type that has a corresponding Variant type.
-For each value in the data, at most one of the `typed_value` and `variant_value` may be non-null.
-A writer may omit either field, which is equivalent to all rows being null.
+If both fields are non-null and either is not an object, the value is invalid. Readers must either fail or return the `typed_value`.
 
-Dictionary IDs in a `variant_value` field refer to entries in the top-level `metadata` field.
+If a Variant is missing in a context where a value is required, readers must either fail or return a Variant null: basic type 0 (primitive) and physical type 0 (null).
+For example, if a Variant is required (like `measurement` above) and both `value` and `typed_value` are null, the returned `value` must be `00` (Variant null).
 
-For an `object`, a null field means that the field does not exist in the reconstructed Variant object.
-All elements of an `array` must be non-null, since array elements cannote be missing.
+### Shredded Value Types
 
-| typed_value | variant_value | Meaning |
-|-------------|----------------|---------|
-| null | null | Field is Variant Null (not missing) in the reconstructed Variant. |
-| null | non-null | Field may be any type in the reconstructed Variant. |
-| non-null | null | Field has this column’s type in the reconstructed Variant. |
-| non-null | non-null | Invalid |
+Shredded values must use the following Parquet types:
 
-The `typed_value` may be absent from the Parquet schema for any field, which is equivalent to its value being always null (in which case the shredded field is always stored as a Variant binary).
-By the same token, `variant_value` may be absent, which is equivalent to their value being always null (in which case the field will always have the value Null or have the type of the `typed_value` column).
+| Variant Type                | Equivalent Parquet Type      |
+|-----------------------------|------------------------------|
+| boolean                     | BOOLEAN                      |
+| int8                        | INT(8, signed=true)          |
+| int16                       | INT(16, signed=true)         |
+| int32                       | INT32 / INT(32, signed=true) |
+| int64                       | INT64 / INT(64, signed=true) |
+| float                       | FLOAT                        |
+| double                      | DOUBLE                       |
+| decimal4                    | DECIMAL(precision, scale)    |
```

Comment:
right now writing int32 values for decimals is optional, do we want to be more proscriptive here (I think its fine to not be, just wanted to call it out).

Reply:
Yes. I'll update the `decimal4` and `decimal8` options. I'll leave the `decimal16` as-is because implementations are still free to use a binary or fixed type (or int32 and int64 if there is some guarantee values would fit).